### DISCUSSION TITLE
How to inherit from multiple class with private attributes?

### DISCUSSION THREAD

**Discussion by TCampmany** [pydantic/pydantic#3480] [Question]
Hi,

I'm trying to create a child class with multiple parents, for my model, and it works really well up to the moment that I add private attributes to the parent classes.

So, this works:

```
from pydantic import BaseModel, PrivateAttr

python
class A(BaseModel):
    not_private_a: str

class B(BaseModel):
    not_private_b: str

class C(B, A):
    ...
```

If only one of my parents have a private attribute, it works too:
```
from pydantic import BaseModel, PrivateAttr

python
class A(BaseModel):
    not_private_a: str
    _private_a: str = PrivateAttr(None)

class B(BaseModel):
    not_private_b: str

class C(B, A):
    ...
```

BUT, if I have private attributes on multiple parents then I have problems...
```
from pydantic import BaseModel, PrivateAttr

python
class A(BaseModel):
    not_private_a: str
    _private_a: str = PrivateAttr(None)

class B(BaseModel):
    not_private_b: str
    _private_b: str = PrivateAttr(None)

class C(B, A):
    ...

Traceback (most recent call last):
  File "/Users/..../PycharmProjects/sandbox/test.py", line 41, in <module>
    class C(B, A, BaseModel, extra=Extra.forbid):
  File "pydantic/main.py", line 367, in pydantic.main.ModelMetaclass.__new__
  File "/usr/local/bin/../Cellar/python@3.8/3.8.12_1/bin/../Frameworks/Python.framework/Versions/3.8/lib/python3.8/abc.py", line 85, in __new__
    cls = super().__new__(mcls, name, bases, namespace, **kwargs)
TypeError: multiple bases have instance lay-out conflict
```

Am I doing something wrong? Is it a known problem? Do we have a workaround?

**Comment by tylerflex** [comment]
@TCampmany did you ever find a workaround? Dealing with the same issue. I believe it is because python doesn't know how to construct the `__slots__` needed to store the private attributes from each model. I noticed that I can run code fine in python 3.6 and 3.9, but I get the same error as you using 3.7 (and possibly 3.8 although I haven't tested).  Perhaps you can try a different python version?  But would be best if there was a more long term solution. It's my understanding that multiple inheritance is not always recommended generally, so I think refactoring of the model inheritance structure might be the ultimate solution..
